src/audio/google/google_rtc_audio_processing.c

```diff
@@ -97,33 +97,36 @@ struct google_rtc_audio_processing_comp_data {
 
 #if CONFIG_GOOGLE_RTC_AUDIO_PROCESSING_MIC_BITS == 16
 typedef int16_t mic_sample_t;
-#define MIC_SCALE ((float)SHRT_MAX)
+#define MIC_SCALE ((float)-SHRT_MIN)
 #else
 typedef int32_t mic_sample_t;
-#define MIC_SCALE ((float)INT_MAX)
+#define MIC_SCALE ((float)-INT_MIN)
 #endif
 
 #if CONFIG_GOOGLE_RTC_AUDIO_PROCESSING_REF_BITS == 16
 typedef int16_t ref_sample_t;
-#define REF_SCALE ((float)SHRT_MAX)
+#define REF_SCALE ((float)-SHRT_MIN)
 #else
 typedef int32_t ref_sample_t;
-#define REF_SCALE ((float)INT_MAX)
+#define REF_SCALE ((float)-INT_MIN)
 #endif
 
 static inline float mic_to_float(mic_sample_t x)
 {
-	return (1.0f / MIC_SCALE) * (float)x;
+	return (1.0f / MIC_SCALE) * x;
 }
 
 static inline mic_sample_t float_to_mic(float x)
 {
-	return (mic_sample_t)(MIC_SCALE * x);
+	float min = -1.0f;
+	float max = 1.0f - 1.0f / MIC_SCALE;
```

Comment:
these conversion utilities seem very generic, like they don't belong in the wrapper.

Reply:
I took a look, there's nothing elsewhere in SOF that matches this convention.  It's more subtle than you'd think, given the requirement for saturation and the (odd to my systems hacker eyes) off-by-one magnitude scaling between positive and negative numbers.  The convention I genuinely thought was the "obvious" one was actually wrong.

I guess maybe let @lkoenig make the call?  My gut says we should keep this as an "AEC-defined internal format" and leave it here, but maybe it's a more common thing that we could move somewhere else.

Reply:
The conversion is "standard" for going from float -1 to 1 to int16_t.
One can ad dithering and rounding eventually. They should maybe be moved into a utility function and unit tested.

Reply:
@andyross did you spot this? https://github.com/thesofproject/sof/blob/b3b9abffe7ee85bf0884d392d3d243161efadbbe/src/include/sof/audio/format.h#L71